docs/basics/api-object.md

```diff
@@ -10,3 +10,50 @@ It is possible to customize the prefix by providing a string value to the
 `--class-prefix` flag, or by passing `--no-class-prefix`. (Note: if no prefix is
 used for k8s resources, this may lead to conflicts - see
 https://github.com/cdk8s-team/cdk8s/issues/140).
+
+### Resource Names
+
+Every object has a name that is autogenerated when the cdk8s app is synthesized.
+The name is the passed in string you assign to an object, prefixed by the name of the
+chart it is contained in, and suffixed by the hash of the construct path.
+
+For example, the following code:
+
+```ts
+import { App, Chart } from 'cdk8s';
+import * as kplus from 'cdk8s-plus-25';
+
+const app = new App();
+const chart = new MyChart(app, 'my-chart');
+
+new kplus.Deployment(chart, 'deployment', {
+  containers: [{ image: 'nginx' }]
+});
+
+app.synth();
+```
+
+Generates a `Deployment` with the name: `my-chart-deployment-c8c354dd`.
+
+Hashes are used to ensure that all resources have unique identifiers. However, they are
+not required and it may sometimes be undesirable to include the hash suffixes.
+
+To remove hashes from resource names, you can change the above code by setting the
+`ChartProps.disableResourceNameHashes` value to `true` like so:
+
+```ts
+import { App, Chart } from 'cdk8s';
+import * as kplus from 'cdk8s-plus-25';
+
+const app = new App();
+const chart = new MyChart(app, 'my-chart', {
+    disableResourceNameHashes: true,
+});
+
+new kplus.Deployment(chart, 'deployment', {
+  containers: [{ image: 'nginx' }]
+});
+
+app.synth();
+```
+Which would similarly generate a `Deployment` now with the name: `my-chart-deployment`.
```

Comment:
Have we mentioned anywhere what issues can non unique identifier cause for the customer?

Reply:
Added a line about how unique names are required in K8s - let me know if this is sufficient